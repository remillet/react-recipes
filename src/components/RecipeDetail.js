import React from 'react';
import { Link } from 'react-router-dom';
import classNames from 'classnames';
import PropTypes from 'prop-types';
import md5 from 'md5';

const RecipeDetail = (props) => {
  const {
    details, style, className, linkTo,
  } = props;

  if (details == null || details.id === '-1') {
    return (
      <p
        style={style}
        className={
          classNames('h3 p2 bg-white italic center', className)
        }
      >
        Select a recipe on the left to see the details.
      </p>
    );
  }

  return (
    <div
      style={style}
      className={
        classNames('h3 p2 bg-white', className)
      }
    >
      <h2 className="h2 center">
        {details.name}
      </h2>
      <img alt="Food" className="fit" src={details.image} />
      <div>
        <span>Category: {details.category}</span>
        <br />
        <span>Calories: {details.calories}</span>
      </div>
      <h3>Ingredients</h3>
      <ul>
        {
          details.ingredients && details.ingredients.map(ingredient => (
            <li key={md5(ingredient)}>{ingredient}</li>
          ))
        }
      </ul>
      <h3>Cooking Instructions</h3>
      <ol>
        {
          details.steps && details.steps.map(step => (
            <li key={md5(step)}>{step}</li>
          ))
        }
      </ol>
      <Link to={linkTo}>
        More info...
      </Link>
    </div>
  );
};

RecipeDetail.defaultProps = {
  details: null,
  linkTo: '/',
  style: null,
  className: null,
};

RecipeDetail.propTypes = {
  linkTo: PropTypes.string,
  style: PropTypes.object,
  className: PropTypes.string,
  details: PropTypes.object,
};

export default RecipeDetail;
