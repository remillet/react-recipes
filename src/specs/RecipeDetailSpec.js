import React from 'react';
import { BrowserRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import RecipeDetail from '../components/RecipeDetail';


describe('<RecipeList />', () => {
  let testRecipeDetail;

  beforeEach(() => {
    testRecipeDetail = {
      id: 1,
      name: 'Recipe list item one',
      category: 'Dessert',
      calories: 180,
      image: 'http:test.me.com/100.jpg',
      ingredients: [
        'Flour',
        'Milk',
        'Butter',
      ],
      steps: [
        'Sift the flour',
        'Scorch the milk',
        'Whip the butter',
      ],
    };
  });

  test('Should not break if no recipe is passed/provided.', () => {
    const component = renderer.create(
      <BrowserRouter>
        <RecipeDetail />
      </BrowserRouter>,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('Should render correctly', () => {
    const component = renderer.create(
      <BrowserRouter>
        <RecipeDetail details={testRecipeDetail} />
      </BrowserRouter>,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('Should render correctly with no ingredients', () => {
    delete testRecipeDetail.ingredients;
    const component = renderer.create(
      <BrowserRouter>
        <RecipeDetail details={testRecipeDetail} />
      </BrowserRouter>,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('Should render correctly with no steps', () => {
    delete testRecipeDetail.steps;
    const component = renderer.create(
      <BrowserRouter>
        <RecipeDetail details={testRecipeDetail} />
      </BrowserRouter>,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
