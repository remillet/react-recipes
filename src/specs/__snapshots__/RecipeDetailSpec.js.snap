// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RecipeList /> Should not break if no recipe is passed/provided. 1`] = `
<p
  className="h3 p2 bg-white italic center"
  style={null}
>
  Select a recipe on the left to see the details.
</p>
`;

exports[`<RecipeList /> Should render correctly 1`] = `
<div
  className="h3 p2 bg-white"
  style={null}
>
  <h2
    className="h2 center"
  >
    Recipe list item one
  </h2>
  <img
    alt="Food"
    className="fit"
    src="http:test.me.com/100.jpg"
  />
  <div>
    <span>
      Category: 
      Dessert
    </span>
    <br />
    <span>
      Calories: 
      180
    </span>
  </div>
  <h3>
    Ingredients
  </h3>
  <ul>
    <li>
      Flour
    </li>
    <li>
      Milk
    </li>
    <li>
      Butter
    </li>
  </ul>
  <h3>
    Cooking Instructions
  </h3>
  <ol>
    <li>
      Sift the flour
    </li>
    <li>
      Scorch the milk
    </li>
    <li>
      Whip the butter
    </li>
  </ol>
  <a
    href="/"
    onClick={[Function]}
  >
    More info...
  </a>
</div>
`;

exports[`<RecipeList /> Should render correctly with no ingredients 1`] = `
<div
  className="h3 p2 bg-white"
  style={null}
>
  <h2
    className="h2 center"
  >
    Recipe list item one
  </h2>
  <img
    alt="Food"
    className="fit"
    src="http:test.me.com/100.jpg"
  />
  <div>
    <span>
      Category: 
      Dessert
    </span>
    <br />
    <span>
      Calories: 
      180
    </span>
  </div>
  <h3>
    Ingredients
  </h3>
  <ul />
  <h3>
    Cooking Instructions
  </h3>
  <ol>
    <li>
      Sift the flour
    </li>
    <li>
      Scorch the milk
    </li>
    <li>
      Whip the butter
    </li>
  </ol>
  <a
    href="/"
    onClick={[Function]}
  >
    More info...
  </a>
</div>
`;

exports[`<RecipeList /> Should render correctly with no steps 1`] = `
<div
  className="h3 p2 bg-white"
  style={null}
>
  <h2
    className="h2 center"
  >
    Recipe list item one
  </h2>
  <img
    alt="Food"
    className="fit"
    src="http:test.me.com/100.jpg"
  />
  <div>
    <span>
      Category: 
      Dessert
    </span>
    <br />
    <span>
      Calories: 
      180
    </span>
  </div>
  <h3>
    Ingredients
  </h3>
  <ul>
    <li>
      Flour
    </li>
    <li>
      Milk
    </li>
    <li>
      Butter
    </li>
  </ul>
  <h3>
    Cooking Instructions
  </h3>
  <ol />
  <a
    href="/"
    onClick={[Function]}
  >
    More info...
  </a>
</div>
`;
